corehq/apps/domain/management/commands/delete_domain.py

```diff
@@ -28,22 +39,53 @@ def handle(self, domain_name, **options):
             print("FYI: There are multiple domain objects for this domain"
                   "and they will all be soft-deleted.")
         if not options['noinput']:
-            confirm = input(
-                """
-                Are you sure you want to delete the domain "{}" and all of it's data?
+            confirm = input(textwrap.dedent(
+                f"""
+                Are you sure you want to delete the domain "{domain_name}" and all of it's data?
                 This operation is not reversible and all forms and cases will be PERMANENTLY deleted.
 
                 Type the domain's name again to continue, or anything else to cancel:
-                """.format(domain_name)
-            )
+                """
+            ))
             if confirm != domain_name:
                 print("\n\t\tDomain deletion cancelled.")
                 return
-        print("Soft-Deleting domain {} "
+        self.hard_delete_cases(domain_name)
+        self.hard_delete_forms(domain_name)
+        print(f"Soft-Deleting domain {domain_name} "
               "(i.e. switching its type to Domain-Deleted, "
-              "which will prevent anyone from reusing that domain)"
-              .format(domain_name))
+              "which will prevent anyone from reusing that domain)")
         for domain_obj in domain_objs:
             assert domain_obj.name == domain_name  # Just to be really sure!
             domain_obj.delete(leave_tombstone=True)
         print("Operation completed")
+
+    def hard_delete_cases(self, domain_name):
+        print("Hard-deleting cases...")
+        case_ids = iter_ids(CommCareCase, 'case_id', domain_name)
+        for chunk in chunked(case_ids, 1000, list):
+            CommCareCase.objects.hard_delete_cases(domain_name, chunk)
+
+    def hard_delete_forms(self, domain_name):
+        print("Hard-deleting forms...")
+        form_ids = iter_ids(XFormInstance, 'form_id', domain_name)
+        for chunk in chunked(form_ids, 1000, list):
+            XFormInstance.objects.hard_delete_forms(domain_name, chunk)
+
+
+def iter_ids(model_class, field, domain, chunk_size=1000):
+    where = Q(domain=domain)
+    rows = paginate_query_across_partitioned_databases(
+        model_class,
+        where,
+        values=[field],
+        load_source='couch_to_sql_migration',
```

Comment:
What's this? It looks like maybe a tag for monitoring. If so, should it be something like `delete_domain`?

Reply:
Haha, copy/paste error. https://github.com/dimagi/commcare-hq/pull/31212/commits/860d724199cf43dfa3d2145ab152ce6cae517c73